homework.cls

```diff
@@ -144,41 +146,37 @@
 % Problem, parts, solution, and claim environments %
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
-% Problem environment
+\RequirePackage[dvipsnames]{xcolor}
+\def\@problemColor{white}
 \ifHW@boxes
-  \RequirePackage{framed}
+  \def\@problemColor{\HW@color}
 \fi
-\theoremstyle{definition}
-\newtheorem{@problem}{\sffamily\@problemname}
-\numberwithin{equation}{@problem}
-\numberwithin{table}{@problem}
-\numberwithin{figure}{@problem}
+
+% Problem environment
+\RequirePackage{varwidth}
+\RequirePackage{tcolorbox}
+\tcbuselibrary{theorems}
+\tcbuselibrary{skins}
+
+\newtcbtheorem{prob}{\@problemname}{enhanced, colback=\@problemColor!5, colframe=\@problemColor!50!white, coltitle=black,fonttitle=\bfseries\sffamily, colbacktitle=\@problemColor!15!white, attach boxed title to top left={yshift=-2mm,xshift=-5mm}, varwidth boxed title, borderline={0.5mm}{0mm}{\@problemColor!15!white}, borderline={0.5mm}{0mm}{\@problemColor!50!white, dashed}, boxed title style={boxrule=1pt}}{prob}
```

Comment:
Something here is responsible for the different default behavior with `boxes` disabled (I suspect!). This whole declaration could possibly be made inside an `\ifHW@boxes`, with the old implementation in the `else` branch.

Reply:
It's the `\@problemColor`! When the `boxes` option isn't used the background color is set to white, and hence it doesn't look like there is any box.